### PR TITLE
SONARGITUB-26 set status even when unexpected errors

### PR DESCRIPTION
**Pull request by janos-ss**

### CODE REVIEW

File: src/main/java/org/sonar/plugins/github/PullRequestIssuePostJob.java [hunk 1]
```diff
@@ -56,15 +61,21 @@ public void describe(PostJobDescriptor descriptor) {
...
-    pullRequestFacade.deleteOutdatedComments();
+      pullRequestFacade.deleteOutdatedComments();
 
-    pullRequestFacade.createOrUpdateGlobalComments(report.hasNewIssue() ? report.formatForMarkdown() : null);
+      pullRequestFacade.createOrUpdateGlobalComments(report.hasNewIssue() ? report.formatForMarkdown() : null);
 
-    pullRequestFacade.createOrUpdateSonarQubeStatus(report.getStatus(), report.getStatusDescription());
+      pullRequestFacade.createOrUpdateSonarQubeStatus(report.getStatus(), report.getStatusDescription());
+    } catch (Exception e) {
+      String msg = "unexpected error occurred while posting comments";
```
Comment (henryju) [inline src/main/java/org/sonar/plugins/github/PullRequestIssuePostJob.java]: We usually start log messages with a capital letter.

File: src/test/java/org/sonar/plugins/github/PullRequestIssuePostJobTest.java [hunk 2]
```diff
@@ -225,4 +225,12 @@ public void testPullRequestAnalysisWithNewBlockerAndCriticalIssues() {
 
     verify(pullRequestFacade).createOrUpdateSonarQubeStatus(GHCommitState.ERROR, "SonarQube reported 2 issues, with 1 critical and 1 blocker");
   }
+
+  @Test
+  public void should_update_sonarqube_status_even_if_unexpected_errors_were_raised() {
+    // raises an NPE due to improper mock setup; not realistic, but good enough for this test
+    pullRequestIssuePostJob.execute(context);
+
+    verify(pullRequestFacade).createOrUpdateSonarQubeStatus(GHCommitState.ERROR, "unexpected error occurred while posting comments");
```
Comment (henryju) [inline src/test/java/org/sonar/plugins/github/PullRequestIssuePostJobTest.java]: This is a bit too much technical for end users on GitHub. I would go for something like "SonarQube failed to complete review of this pull request". @bellingard WDYT?

Comment (ofirnk) [inline src/test/java/org/sonar/plugins/github/PullRequestIssuePostJobTest.java]: I'd suggest to leverage the `details` tag for extra details:

<details>
  <summary>Click for error details</summary>
  whatever
</details>

---------------------
source markdown / html:
```
<details>
  <summary>Click for error details</summary>
  whatever
</details>
```

Comment (bellingard) [inline src/test/java/org/sonar/plugins/github/PullRequestIssuePostJobTest.java]: @henryju Yes, your suggestion is more user friendly indeed. For what @ofirnk suggests, if we can add more context, this is a nice idea to use the "details" tag.

Comment (janos-ss) [inline src/test/java/org/sonar/plugins/github/PullRequestIssuePostJobTest.java]: Thanks @ofirnk, looks cool, unfortunately it doesn't seem to work for the commit status message, the html will be displayed verbatim

Comment (ofirnk) [inline src/test/java/org/sonar/plugins/github/PullRequestIssuePostJobTest.java]: Missed that it's the details message.. 

> Just to make sure - we're talking about this message, right?
> image
> (The highlighted part)

So I'd suggest to **_comment on the PR_** with the (hidden) details and the message. That's something that IMO needs attention by all PR subscribers: _An essential check has failed_.
Also, that would benefit much from the error context .

**Pull request merged**